test/physics/em/MollerBhabha.test.cc

```diff
@@ -106,184 +108,201 @@ class MollerBhabhaInteractorTest : public celeritas_test::InteractorHostTestBase
-        this->sanity_check(result);
-
-        sampled_inc_exit_dir.push_back(result.direction);
-        sampled_inc_exit_energy.push_back(result.energy.value());
-        sampled_inc_energy_dep.push_back(result.energy_deposition.value());
-        auto secondary = result.secondaries[0];
-        sampled_sec_dir.push_back(secondary.direction);
-        sampled_sec_energy.push_back(secondary.energy.value());
-    }
-
-    //// PRIMARY
-    // Incident particle's exiting directions after each interactor call
-    Real3 expect_inc_exit_dir[4] = {{-5.98339146337106900e-02,
-                                     1.18556847856613259e-02,
-                                     9.98137939063468704e-01},
-                                    {6.19406177347736359e-03,
-                                     1.89364553704606886e-03,
-                                     9.99979023632659225e-01},
-                                    {-7.35997559188817274e-05,
-                                     1.63669752943484403e-06,
-                                     9.99999997290198617e-01},
-                                    {-3.22691294854657801e-07,
-                                     3.84815259626114375e-08,
-                                     9.99999999999947153e-01}};
-
-    // Incident particle's final energy after each interactor call [MeV]
-    double expect_inc_exit_energy[4] = {9.92665477497147508e-01,
-                                        9.99547740388010375e+00,
-                                        9.99994691659400587e+02,
-                                        9.99999989666164765e+04};
-
-    // Incident particle's deposited energy after each interactor call [MeV]
-    double expect_inc_energy_dep[4] = {0, 0, 0, 0};
-
-    //// SECONDARY
-    // Secondary directions for each interactor call
-    Real3 expect_sec_dir[4] = {{9.73837231712127771e-01,
-                                -1.92959249322042975e-01,
-                                1.20032388263890455e-01},
-                               {-9.53982991975042882e-01,
-                                -2.91651213894371375e-01,
-                                6.96851523373852172e-02},
-                               {9.97163856758067957e-01,
-                                -2.21747422993688603e-02,
-                                7.19202584765359088e-02},
-                               {9.92462814088369005e-01,
-                                -1.18353002253964334e-01,
-                                3.17825346577122123e-02}};
-
-    // Secondary energies for each interactor call [MeV]
-    real_type expect_sec_energy[4] = {7.33452250285249879e-03,
-                                      4.52259611989608724e-03,
-                                      5.30834059944587742e-03,
-                                      1.03338351935111108e-03};
-
-    for (int i : celeritas::range(4))
+    for (double inc_e : {1.0, 10.0, 100.0, 1000.0})
```

Comment:
Since the model is good down to at least 1 keV, could you extend this to 1e-3? Also, you're only testing with incident electrons, so another loop over particle type would be good as well.

Reply:
Given the fact that the maximum energy transfer fraction for Moller is 0.5, the lower energy limit for the incident particle becomes `E > 2e-3`. For Bhabha, the maximum energy transfer fraction is 1, so in that case `E > 1e-3`.

A possible issue (which may lead us to add a hard limit to the ` do { ... } while ( ... )` loop) is that as `E` approaches the lower limit, the number engine samples goes to infinity (see `expected_avg_engine_samples` numbers in the test harness).

Reply:
Just saw this after making my comment below. Good point about the lower transfer energy. Looking at the exported data tables, looks like the Moller cross sections go down to just under 2 keV and Bhabha just under 1 keV. (At those values, the cross section is zero.) So we should definitely adjust the lower bounds for the electron applicability in the Model.